core/state/trie_prefetcher.go

```diff
@@ -77,15 +112,86 @@ func newTriePrefetcher(db Database, root common.Hash, namespace string) *triePre
 		storageWasteMeter: metrics.GetOrRegisterMeter(prefix+"/storage/waste", nil),
 	}
 	go p.abortLoop()
+	go p.mainLoop()
 	return p
 }
 
+func (p *triePrefetcher) mainLoop() {
+	for {
+		select {
+		case <-p.copyChan:
+			copy := &triePrefetcher{
+				db:      p.db,
+				root:    p.root,
+				fetches: make(map[common.Hash]Trie, len(p.fetchers)),
+			}
+			// we're copying an active fetcher, retrieve the current states
+			for root, fetcher := range p.fetchers {
+				copy.fetches[root] = fetcher.peek()
+			}
+			p.copyDoneChan <- copy
+
+		case pMsg := <-p.prefetchChan:
+			fetcher := p.fetchers[pMsg.root]
+			if fetcher == nil {
+				fetcher = newSubfetcher(p.db, pMsg.root, pMsg.accountHash)
+				p.fetchers[pMsg.root] = fetcher
+			}
+			fetcher.schedule(pMsg.keys)
+
+		case tireMsg := <-p.trieChan:
+			fetcher := p.fetchers[tireMsg.root]
```

Comment:
as the original way, when trie be used, the `prefetcher` on this root should be stoped, as it would reach downwards to disk until get a trie in the `caller`

Reply:
yes, it is ok, the sub fetcher of the root will be stopped, but not in mainLoop.
Deliberated moved outside for performance concern.